### PR TITLE
fix: make image endpoint highest priority

### LINKED ISSUE #12589: [5.0 Beta] catchall route matches _image endpoint in Node
### Astro Info

```block
Astro                    v5.0.0-beta.12
Node                     v22.11.0
System                   macOS (arm64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/node
Integrations             @astrojs/tailwind
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

We get 404 errors for all remote images on the production build on our servers. Locally everything works fine.
CleanShot 2024-12-03 at 09 56 55 2@2x

This is our image component:

````
---
import { Image } from 'astro:assets';
const { image, loading, widths, sizes } = Astro.props;

const imageWidths = widths ?? [240, 360, 480, 720, 960, 1440, 1820, 2560, 3840];
const imageSizes = sizes ?? '100vw';
---

{
	image && (image.src || image.url) && (
		<Image
			src={image.src || image.url}
			alt={image.alt ?? ''}
			width={image.width}
			height={image.height}
			class={Astro.props.class}
			widths={imageWidths}
			sizes={imageSizes}
			loading={loading ?? 'lazy'}
		/>
	)
}
````

this is the astro.config.mjs

````
import { defineConfig } from 'astro/config';

import node from '@astrojs/node';

import tailwind from '@astrojs/tailwind';

// https://astro.build/config
export default defineConfig({
	output: 'server',

	adapter: node({
		mode: 'standalone',
	}),

	integrations: [
		tailwind({
			nesting: true,
		}),
	],

	image: {
		domains: ['cms.fooo.com', 'cms.fooo.com.ddev.site'],
	},
});
````

### What's the expected result?

All images should be loaded correctly, despite the NODE_ENV mode.

### Link to Minimal Reproducible Example

[Github Repo](https://github.com/MathiasGmeiner/astro-beta)

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by ascorbic**
## Changes

Ensures that the image endpoint is highest priority, by adding it to the start of the route array rather than the end

## Testing

Tested locally with repro site

Fixes #12589. Thanks @MathiasGmeiner

### CODE REVIEW

**Review by florian-lefebvre — APPROVED** [review]

**Comment by ascorbic** [comment]
!preview image-endpoint

File: packages/astro/src/assets/endpoint/config.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function ensureImageEndpointRoute(
 	cwd?: string,
 ) {
 	if (!manifest.routes.some((route) => route.route === '/_image')) {
```
[the remark this answers is not in the document]
Comment (ascorbic) [reply packages/astro/src/assets/endpoint/config.ts:25]: Good point. A different issue but we can include it in this

**Review by ematipico — APPROVED** [review]

**Review by Princesseuh — APPROVED** [review]

**Review by ematipico — APPROVED** [review]
I checked out the branch, and confirmed that the tests passed on my machine too

**Comment by ascorbic** [comment]
I'm going to wait til they pass in CI before merging

**Pull request merged**